src/ui/components/search/autocompletecomponent.js

```diff
@@ -214,7 +214,7 @@ export default class AutoCompleteComponent extends Component {
       resultIndex: this._resultIndex,
       promptHeader: this._originalQuery.length === 0 ? this.promptHeader : null,
       listLabelIdName: this.listLabelIdName,
-      autocompleteContainerIdName: `yxt-AutoComplete-container-${this._config.name.replaceAll('.', '-')}`,
+      autocompleteContainerIdName: `yxt-AutoComplete-container-${this._config.name?.replaceAll('.', '-')}`,
```

Comment:
I think you need a question mark after `replaceAll` as well like like `this._config.name?.replaceAll?('.', '-')`

Reply:
i thought if this._config.name is null or undefined, it will just return `yxt-AutoComplete-container-`

Reply:
oh, you're correct. The extra ? after `replaceAll` would only be useful if name is present but for some reason it doesn't have the replaceAll method